File: app/src/main/java/com/montreal/wtm/model/Speaker.java
package com.montreal.wtm.model;


import android.os.Parcel;
import android.os.Parcelable;

import com.google.firebase.database.IgnoreExtraProperties;
import com.google.firebase.database.PropertyName;

@IgnoreExtraProperties
public class Speaker implements Parcelable {

    @PropertyName("name")
    public String name;

    @PropertyName("title")
    public String title;
    
    @PropertyName("shortBio")
    public String shortBio;
    
    @PropertyName("bio")
    public String bio;
    
    @PropertyName("photoUrl")
    public String photoUrl;

    
    public Speaker() {
    }

    public Speaker(String name, String title, String shortBio, String bio, String photoUrl) {
        this.name = name;
        this.title = title;
        this.shortBio = shortBio;
        this.bio = bio;
        this.photoUrl = photoUrl;
    }

    public String getName() {
        return name;
    }

    public String getShortBio() {
        return shortBio;
    }

    public String getBio() {
        return bio;
    }

    public String getPhotoUrl() {
        return photoUrl;
    }

    public String getTitle() {
        return title;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.name);
        dest.writeString(this.title);
        dest.writeString(this.shortBio);
        dest.writeString(this.bio);
        dest.writeString(this.photoUrl);
    }

    protected Speaker(Parcel in) {
        this.name = in.readString();
        this.title = in.readString();
        this.shortBio = in.readString();
        this.bio = in.readString();
        this.photoUrl = in.readString();
    }

    public static final Creator<Speaker> CREATOR = new Creator<Speaker>() {
        @Override
        public Speaker createFromParcel(Parcel source) {
            return new Speaker(source);
        }

        @Override
        public Speaker[] newArray(int size) {
            return new Speaker[size];
        }
    };
}

File: app/src/main/java/com/montreal/wtm/model/Speaker.kt
package com.montreal.wtm.model


import android.os.Parcel
import android.os.Parcelable

import com.google.firebase.database.PropertyName


class Speaker : Parcelable {

  val NO_SESSION = -1

  @PropertyName("name")
  lateinit var name: String

  @PropertyName("title")
  lateinit var title: String

  @PropertyName("shortBio")
  lateinit var shortBio: String

  @PropertyName("bio")
  lateinit var bio: String

  @PropertyName("photoUrl")
  lateinit var photoUrl: String

  @PropertyName("id")
  var id: Int = 0

  @PropertyName("socials")
  var socials: ArrayList<Social> = ArrayList()


  @PropertyName("sessionId")
  var sessionId: Int = NO_SESSION

  constructor() {}

  override fun describeContents(): Int {
    return 0
  }

  override fun writeToParcel(dest: Parcel, flags: Int) {
    dest.writeInt(this.id)
    dest.writeString(this.name)
    dest.writeString(this.title)
    dest.writeString(this.shortBio)
    dest.writeString(this.bio)
    dest.writeString(this.photoUrl)
    dest.writeSerializable(this.socials)
    dest.writeInt(this.sessionId)
  }

  protected constructor(`in`: Parcel) {
    this.id = `in`.readInt() ?: 0
    this.name = `in`.readString() ?: ""
    this.title = `in`.readString() ?: ""
    this.shortBio = `in`.readString() ?: ""
    this.bio = `in`.readString() ?: ""
    this.photoUrl = `in`.readString() ?: ""
    this.socials = `in`.readSerializable() as ArrayList<Social>
    this.sessionId = `in`.readInt()
  }

  companion object {

    @JvmField
    val CREATOR: Parcelable.Creator<Speaker> = object : Parcelable.Creator<Speaker> {
      override fun createFromParcel(source: Parcel): Speaker {
        return Speaker(source)
      }

      override fun newArray(size: Int): Array<Speaker?> {
        return arrayOfNulls(size)
      }
    }
  }
}
